presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadata.java

```diff
@@ -1307,16 +1307,26 @@ public OptionalLong metadataDelete(ConnectorSession session, ConnectorTableHandl
             throw new TableNotFoundException(handle.getSchemaTableName());
         }
 
+        HiveBasicStatistics deletedDataStatistics;
         if (table.get().getPartitionColumns().isEmpty()) {
+            deletedDataStatistics = HiveBasicStatistics.createFromPartitionParameters(table.get().getParameters());
             metastore.truncateUnpartitionedTable(session, handle.getSchemaName(), handle.getTableName());
         }
         else {
+            int deletedPartitionsCount = 0;
+            deletedDataStatistics = createZeroStatistics();
             for (HivePartition hivePartition : getOrComputePartitions(layoutHandle, session, tableHandle)) {
-                metastore.dropPartition(session, handle.getSchemaName(), handle.getTableName(), toPartitionValues(hivePartition.getPartitionId()));
+                List<String> partitionValues = toPartitionValues(hivePartition.getPartitionId());
+                Partition partition = metastore.getPartition(handle.getSchemaName(), handle.getTableName(), partitionValues)
+                        .orElseThrow(() -> new PartitionNotFoundException(handle.getSchemaTableName(), partitionValues));
+                metastore.dropPartition(session, handle.getSchemaName(), handle.getTableName(), partitionValues);
+                deletedPartitionsCount++;
+                deletedDataStatistics = deletedDataStatistics.add(HiveBasicStatistics.createFromPartitionParameters(partition.getParameters()));
             }
+            metastore.finishDeleteFromExistingTable(session, handle.getSchemaName(), handle.getTableName(), deletedDataStatistics, deletedPartitionsCount);
         }
-        // it is too expensive to determine the exact number of deleted rows
```

Comment:
is it no longer expensive? :)

Does it mean that `DELETE FROM hive.table WHERE partkey = 1` will now report number of deleted rows? (But it will be stats-based, potentially inexact?)

Reply:
Yes